File: app/src/main/java/com/ongouser/Home/ForgotPasswordActivity.java
package com.ongouser.Home;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

import androidx.appcompat.app.AppCompatActivity;

import com.ongouser.Login.LoginActivity;
import com.ongouser.R;

public class ForgotPasswordActivity extends AppCompatActivity {
    Context mContext;
    Button btnSend;
    ImageView ivBack;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_forgot_password);
        mContext=this;
        ivBack=findViewById(R.id.ivBack);
        ivBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
        btnSend=findViewById(R.id.btnSend);
        btnSend.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Go();
            }
        });
    }
    public void Go(){
        Intent intent= new Intent( this, LoginActivity.class);
        startActivity(intent);
    }
}

File: app/src/main/java/com/ongouser/Login/ForgotPasswordActivity.kt
package com.ongouser.Login

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import com.ongouser.R
import com.ongouser.base.BaseActivity
import com.ongouser.manager.restApi.RestObservable
import com.ongouser.manager.restApi.Status
import com.ongouser.pojo.ForgotPasswordResponse
import com.ongouser.utils.others.CommonMethods
import com.ongouser.utils.others.Constants
import com.ongouser.utils.others.Constants.Companion.SecurityKeyValue
import com.ongouser.utils.others.MyApplication
import com.ongouser.viewmodel.AuthViewModel
import kotlinx.android.synthetic.main.activity_forgot_password.*


class ForgotPasswordActivity : BaseActivity(), View.OnClickListener, Observer<RestObservable> {

    private val viewModel: AuthViewModel
            by lazy { ViewModelProviders.of(this).get(AuthViewModel::class.java) }
    lateinit var mContext: ForgotPasswordActivity
    override fun getContentId(): Int {
        return R.layout.activity_forgot_password
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mContext = this

        btnSend.setOnClickListener(mContext)
        ivBack.setOnClickListener(mContext)
    }

    override fun onClick(v: View?) {
        when (v!!.id) {
            R.id.btnSend -> {
                forgotPasswordApi()
                hideKeyboard(btnSend)

            }
            R.id.ivBack -> {
                onLeftIconClick()
            }

        }
    }

    private fun isValid(): Boolean {
        var check = false
        if (!mValidationClass.isNetworkConnected)
            showAlerterRed(resources.getString(R.string.no_internet))
        else if (mValidationClass.checkStringNull(edEmail.text.toString()))
            showAlerterRed(resources.getString(R.string.error_email))
        else if (!mValidationClass.isValidEmail(edEmail.text.toString()))
            showAlerterRed(resources.getString(R.string.error_validemail))
        else
            check = true
        return check
    }

    private fun forgotPasswordApi() {
        if (isValid()) {
            val map = HashMap<String, String>()
            map.put("email", edEmail.text.toString().trim())
            map.put(Constants.SecurityKey, SecurityKeyValue)
//            map.put("device_token", token!!)

            viewModel.forgotPasswordApi(this, true, map)
            viewModel.mResponse.observe(this, this)
        }
    }

    override fun onChanged(it: RestObservable?) {
        when {
            it!!.status == Status.SUCCESS -> {
                if (it.data is ForgotPasswordResponse) {
                    val forgotPasswordResponse: ForgotPasswordResponse = it.data
                    if (forgotPasswordResponse.getCode()!!.equals("200")) {
                        showSuccessToast(mContext, forgotPasswordResponse.getMessage().toString())

                        var intent = Intent(mContext, LoginActivity::class.java)
                        startActivity(intent)
                        finishAffinity()
                    } else {
                      showAlerterRed(forgotPasswordResponse.getMessage().toString())
                    }

                }
            }
            it.status == Status.ERROR -> {
                if (it.data != null) {
                    showAlerterRed(it.data as String)
                } else {
                    showAlerterRed(it.error!!.toString())
                }
            }
            it.status == Status.LOADING -> {

            }
        }
    }

}
